analyses/upstream-analysis/util/function-calculate-qc-metrics.R

```diff
@@ -0,0 +1,405 @@
+
+  # plot percent mitochondrial reads
+  mito_num <- plot_distribution(seurat_obj, 
+                                features = "percent.mito",
+                                grouping = grouping) +
+    theme_Publication(base_size = 11) + 
+    xlab("") + 
+    ylab("") + 
+    scale_fill_manual(values = alpha(c(vln_plot_color_df$hex_codes), .4)) 
+  
+  # get the legend for one of the plots to use as legend for the combined plot
+  legend_grid <- get_legend(mito_num)
+  
+  # Now add the title
+  title_name_grid <- ggdraw() +
+    draw_label(title_name, fontface = "bold", x = 0, hjust = 0) +
+    theme(
+      # add margin on the left of the drawing canvas,
+      # so title is aligned with left edge of first plot
+      plot.margin = margin(0, 0, 0, 7))
+  
+  # Combine plots 
+  plot_row <- plot_grid(genes_num + theme(legend.position = "none"),
+                        umi_num + theme(legend.position = "none"),
+                        mito_num + theme(legend.position = "none"),
+                        legend_grid, 
+                        ncol = 4)
+  
+  plot <- plot_grid(title_name_grid, plot_row, ncol = 1, rel_heights = c(0.1, 1))
+  
+  return(plot)
+}
+
+###############################################################################################################
+#' Function to make QC plots a the given Seurat object
+#' Project_Path is where sub-directories will be created and figures will be saved
+#' @param Seurat_obj
+#' @param Project_Path
+#' @param Figure_Total
+#' @param Analysis_Name
+#' @param File_Name
+#' 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' 
+Generate_QC_Plots_1 <- function(Seurat_obj, Project_Path, Figure_Total, Analysis_Name, File_Name, palette) {
+  
+  # Create directory to save figures to
+  dir.create(paste(Project_Path, "/", Analysis_Name, sep = ""))
+  
+  # Open path/name to write figures to 
+  # pdf(file = paste(Project_Path, "/", Analysis_Name, "/", Analysis_Name, "_", Figure_Total, "_", File_Name, ".pdf", sep = ""), width = length(unique(Seurat_obj$ID))*12, height = 12)
+  fname <- paste0(Project_Path, "/", Analysis_Name, "/", Analysis_Name, "_", Figure_Total, "_", File_Name, ".png", sep = "")
+  print(fname)
+
+  # Make density plots of UMI count, gene count split and colored by sample ID
+  plot1 <- ggplot() + geom_density(data = Seurat_obj@meta.data, mapping = aes(x = nCount_RNA, fill = ID), alpha = 0.2) + 
+    scale_x_log10() + 
+    theme_Publication() + 
+    #scale_fill_manual(values = getPalette(colourCount)) +
+    ggtitle("nCount_RNA vs Cell Density") + 
+    ylab("Cell Density") + 
+    theme(aspect.ratio = 1)
+
+  plot2 <- ggplot() + geom_density(data = Seurat_obj@meta.data, mapping = aes(x = nFeature_RNA, fill = ID), alpha = 0.2) +
+    scale_x_log10() +
+    theme_Publication() + 
+    #scale_fill_manual(values = getPalette(colourCount)) +
```

Comment:
Will the couple of commented out commands in this function and the functions below (e.g. pdf... or scale_fill...) be explained later in the README as possible options to change manually, or are they actually commands that will not be used? It's extremely minor, but if they won't be used, maybe consider removing them, just for readability if anyone digs into the code when they use it.

Reply:
Code lines commented out are recommended to be removed in general, unless they might be useful later. There is also no need to explain that anywhere.